src/Data/Profunctor/Join.purs

```diff
@@ -0,0 +1,21 @@
+module Data.Profunctor.Join where
+
+import Prelude
+
+import Data.Functor.Invariant (class Invariant)
+import Data.Newtype (class Newtype)
+import Data.Profunctor (class Profunctor, dimap)
+
+-- | Turns a `Profunctor` into a `Invariant` functor by equating the two type
+-- | arguments.
+newtype Join p a = Join (p a a)
```

Comment:
I found the previous discussion of this [here](https://github.com/purescript/purescript-monoid/pull/13#issuecomment-79726322).